### PR TITLE
Allow Grain Integrations to update their own configs

### PR DESCRIPTION
**Pull request by topocount**
A grain integration might want to store some details within an instance,
  such as a smart contract address. This update allows arbitrary updates
  to a scoped object within the grain config file.

[Merge Plan](https://github.com/sourcecred/sourcecred/issues/3139#issuecomment-978073574)

Test Plan:

Unit tests are provided at the grain integration interface. Integration
testing will be demonstrated in the test instance using the merkle
integration.

### CODE REVIEW

File: packages/sourcecred/src/core/ledger/grainIntegration.js [hunk 1]
```diff
@@ -46,6 +49,7 @@ export type IntegrationConfig = {|
   // distributed funds via a configured integration
   processDistributions: boolean,
   currency: Currency,
+  integration: ?Object,
```
Comment (blueridger) [inline packages/sourcecred/src/core/ledger/grainIntegration.js:52]: This is a confusing line. Maybe it wants to be renamed `config` or something?

**Review by blueridger — APPROVED** [review]
1 change requested, but otherwise approved

File: packages/sourcecred/src/core/ledger/grainIntegration.js (hunk 1 above)
Comment (topocount) [reply packages/sourcecred/src/core/ledger/grainIntegration.js:52]: This is confusing in the context of the named `IntegrationConfig` type, but I don't think its confusing once within the context of an integration. Within an integration, its pretty clearly just reading its own configuration.

From within the context of a grain integration, we have 
```
{
  accounting,
  currency,
  processDisitributions,
  integration
}
```

I'd be open to renaming it `self` but I think naming it `config` would more confusing and less descriptive in the alternative context, which needs to be much more approachable for devs looking to build their own integrations.

Comment (topocount) [reply packages/sourcecred/src/core/ledger/grainIntegration.js:52]: I've added a comment to clarify what's going on instead.

**Pull request merged**